File: 06/main.swift
import Foundation

func input() -> String {
	let inputString = NSFileHandle.fileHandleWithStandardInput().availableData
	let strData = NSString(data: inputString, encoding: NSUTF8StringEncoding)!
	return strData.stringByTrimmingCharactersInSet(NSCharacterSet.newlineCharacterSet())
}

print("What is your current age?")
let age = Int(input())!
print("At what age would you like to retire?")
let retirementAge = Int(input())!
let currentYear = NSCalendar.currentCalendar().component(.Year, fromDate: NSDate());
let yearsLeft = retirementAge - age
let retirementYear = currentYear + yearsLeft

print("You have \(yearsLeft) years left until you can retire.")
print("It's \(currentYear), so you can retire in \(retirementYear).")

File: 06/main.clj
(println "What is your current age?")
(def age (Integer. (re-find #"\d+" (read-line))))
(println "At what age would you like to retire?")
(def retirement-age (Integer. (re-find #"\d+" (read-line))))
(def current-year (let [cal (java.util.Calendar/getInstance)]
                    (. cal get java.util.Calendar/YEAR)))
(def years-left (- retirement-age age))
(def retirement-year (+ years-left current-year))

(println (str "You have " years-left " years left until you can retire."))
(println (str "It's " current-year ", so you can retire in " retirement-year "."))
